parsl/executors/high_throughput/executor.py

```diff
@@ -483,38 +500,58 @@ def scale_out(self, blocks=1):
         r = []
         for i in range(blocks):
             if self.provider:
-                block = self.provider.submit(self.launch_cmd, 1, 1)
-                logger.debug("Launched block {}:{}".format(i, block))
-                if not block:
+                external_block_id = str(len(self.blocks))
+                launch_cmd = self.launch_cmd.format(block_id=external_block_id)
+                internal_block = self.provider.submit(launch_cmd, 1, 1)
+                logger.debug("Launched block {}->{}".format(external_block_id, internal_block))
+                if not internal_block:
                     raise(ScalingFailed(self.provider.label,
                                         "Attempts to provision nodes via provider has failed"))
-                self.blocks.extend([block])
+                r.extend([external_block_id])
+                self.blocks[external_block_id] = internal_block
             else:
                 logger.error("No execution provider available")
                 r = None
         return r
 
-    def scale_in(self, blocks):
+    def scale_in(self, blocks, block_ids=[]):
```

Comment:
Maybe this should be a single parameter: blocks. That should be either a number N, meaning kill N blocks of parsls choice, or a list of block_ids, meaning kill these blocks.

It never makes sense to pass both or neither - always exactly one (I think)

(It's the sum type Vs product type distinction from type theory if you like that sort of thing...)

Reply:
I'd have preferred having two kwargs `blocks_ids=[<list of block ids], block_count=<int>` but that would mean making changes everywhere to match the name changes everywhere.

I worry that overloading blocks to mean two different things based on type won't be very clear.

Reply:
As the code is now, you have to pass something for the blocks parameter, right? Even if it's going to get ignored because there is a block_ids list?

Reply:
Maybe adding =None to blocks will let it be used both positionally but also omitted if the block_ids keyword is used

Reply:
Maybe adding =None to blocks will let it be used both positionally but also omitted if the block_ids keyword is used